test/test_sparse_ops.py

```diff
@@ -0,0 +1,47 @@
+import math
+from itertools import product, chain
+from numbers import Number
+import random
+
+import unittest
+
+import torch
+
+from torch.testing._internal.common_utils import \
+    (TestCase, run_tests, torch_to_numpy_dtype_dict, suppress_warnings,
+     TEST_NUMPY, IS_MACOS, make_tensor)
+from torch.testing._internal.common_methods_invocations import \
+    (unary_ufuncs)
+from torch.testing._internal.common_device_type import \
+    (instantiate_device_type_tests, ops, dtypes)
+from torch.testing import \
+    (floating_types_and, integral_types, all_types_and_complex_and)
+
+import numpy as np
+
+class TestSparseUnaryUfuncs(TestCase):
+    exact_dtype = True
+
+    @ops((_fn for _fn in unary_ufuncs if _fn.is_sparse_supported is True))
+    def test_output_all_dtypes(self, device, dtype, op):
+        i = torch.tensor([[0, 1, 1],
```

Comment:
Acquire inputs using the op's `sample_inputs`, run these through the op to create an "expected" output, then convert the strided tensors in the sample input to sparse tensors and run those through the op. Then you'll have a sparse tensor output and a strided tensor output, and the values should be the same. Compare them by converting the sparse tensor to a strided (AKA "dense") tensor.

Reply:
This is done, please check the new implementation in test_sparse.py